direct/src/showbase/ShowBase.py

```diff
@@ -371,19 +376,19 @@ def __init__(self, fStartDirect=True, windowType=None):
         #: traverse it automatically in the collisionLoop task, so you won't
         #: need to call :meth:`~panda3d.core.CollisionTraverser.traverse()`
         #: yourself every frame.
-        self.cTrav = 0
-        self.shadowTrav = 0
+        self.cTrav: CollisionTraverser | Literal[0] = 0
+        self.shadowTrav: CollisionTraverser | Literal[0] = 0
         self.cTravStack = Stack()
         # Ditto for an AppTraverser.
-        self.appTrav = 0
+        self.appTrav: CollisionTraverser | Literal[0] = 0
```

Comment:
This is not meant to be a CollisionTraverser, better make it Any (though we really just ought to remove this field)

Reply:
I should note that this usages of this syntax (contrast to `Union[CollisionTraverser, Literal[0]]`) only works on Python 3.10 and below.

Reply:
Unless `from __future__ import annotations` is used (which it is). It will just be a string at runtime, and type-checkers can understand it even if they're run on 3.8 or 3.9.